src/api_manager/check_workflow.cc

```diff
@@ -20,13 +20,16 @@
 #include "src/api_manager/check_service_control.h"
 #include "src/api_manager/fetch_metadata.h"
 #include "src/api_manager/quota_control.h"
+#include "src/api_manager/request_preprocess.h"
 
 using ::google::api_manager::utils::Status;
 
 namespace google {
 namespace api_manager {
 
 void CheckWorkflow::RegisterAll() {
+  // Validate request
+  Register(RequestValidator);
```

Comment:
It seem using "CheckWorkflow" is over kill.   "Workflow" was designed for asynchronize work what require callback, such as fetching token, fetching something.  

Here we just remove a header, doens't need to be done in Workflow.

How about

Introduce a "pre-processing" step in here:  https://github.com/cloudendpoints/esp/blob/master/src/api_manager/request_handler.cc#L41

Reply:
Moved the code to the constructor of the RequestHandler class.